petab/problem.py

```diff
@@ -86,34 +93,48 @@ def from_files(sbml_file: str = None,
             condition_file: PEtab condition table
             measurement_file: PEtab measurement table
             parameter_file: PEtab parameter table
+            visualization_files: PEtab visualization tables
         """
 
         sbml_model = sbml_document = sbml_reader = None
-        condition_df = measurement_df = parameter_df = None
+        condition_df = measurement_df = parameter_df = visualization_df = None
 
         if condition_file:
             condition_df = conditions.get_condition_df(condition_file)
+
         if measurement_file:
             if isinstance(measurement_file, str):
                 measurement_df = measurements.get_measurement_df(
                     measurement_file)
             else:
                 # If there are multiple tables, we will merge them
-                measurement_df = measurements.concat_measurements(
-                    measurement_file)
+                measurement_df = core.concat_tables(
+                    measurement_file, measurements.get_measurement_df)
+
         if parameter_file:
             parameter_df = parameters.get_parameter_df(parameter_file)
+
         if sbml_file:
             sbml_reader = libsbml.SBMLReader()
             sbml_document = sbml_reader.readSBML(sbml_file)
             sbml_model = sbml_document.getModel()
 
+        if visualization_files:
+            if isinstance(visualization_files, str):
+                visualization_df = core.get_visualization_df(
+                    visualization_files)
+            else:
+                # If there are multiple tables, we will merge them
```

Comment:
one could also have the 1-length list inside concat_tables, would make the special case treatment easier.